src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1391,15 +1391,20 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     // Arguments:
     // - trimTrailingWhitespace: enable removing any whitespace from copied selection
     //    and get text to appear on separate lines.
-    void TermControl::CopySelectionToClipboard(bool trimTrailingWhitespace)
+    bool TermControl::CopySelectionToClipboard(bool trimTrailingWhitespace)
     {
-        // extract text from buffer
-        const auto copiedData = _terminal->RetrieveSelectedTextFromBuffer(trimTrailingWhitespace);
+        if (_terminal != nullptr && _terminal->IsAreaSelected())

```

Comment:
```suggestion
        if (_terminal->IsAreaSelected())
```
Is this even possible? Can we have a `TermControl` with no `Terminal`?